### PR TITLE
Image upload service

### PR DESCRIPTION
**Pull request by prakashchoudhary07**

### CODE REVIEW

File: app/components/upload-image.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        const message = data.message;
+        this.handelResponseStatusMessage(status, message);
+      })
+      .catch((err) => {
+        this.setImageUploadSuccess(false);
+        this.setStatusMessage(
+          'Error occured, please try again and if the issue still exists contact administrator and create a issue on the repo with logs'
+        );
+        console.error(err);
+      })
```
Comment (swarajpure) [inline app/components/upload-image.js:84]: Mix of async and fetch 😅 
Can we please keep it consistent?

Comment (prakashchoudhary07) [reply app/components/upload-image.js:84]: Ohh I forgot :sweat:

File: app/components/upload-image.hbs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<div class="image-upload">
+  <div
+    class="drop-area {{if this.overDropZone 'drop-area__highlight'}}"
+    {{on 'drop' this.handleDrop}}
+    {{on 'dragover' this.handleDragOver}}
+    {{on 'dragenter' this.handleDragEnter}}
+    {{on 'dragleave' this.handleDragLeave}}
+  >
+    <form class="image-form">
+      <p class="image-form__text">
```
Comment (ankushdharkar) [inline app/components/upload-image.hbs:10]: Can we nest `<p>` tags?

File: app/components/upload-image.hbs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        {{#if this.isImageSelected}}
+          You have selected{{this.imageFileName}}
+        {{else}}
+          Drag and drop file here
+          <p class="image-form__text">
+            or
+          </p>
+        {{/if}}
```
Comment (ankushdharkar) [inline app/components/upload-image.hbs:18]: What comes after the `or`?

File: app/components/upload-image.hbs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      </p>
+      <input
+        class="image-form__input"
+        type="file"
+        id="image"
+        accept="image/png, image/jpeg"
+        disabled={{this.isImageUploading}}
+        {{on 'change' this.handelBrowseImage}}
```
Comment (ankushdharkar) [inline app/components/upload-image.hbs:26]: `handel` is a typo

File: app/components/upload-image.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  @tracked overDropZone = false;
+  @tracked isImageUploading = false;
+  @tracked imageUploadSuccess = false;
+  @tracked statusMessage;
+  @tracked imageFileName;
+  uploadUrl = this.args.uploadUrl;
+  formKeyName = this.args.formKeyName;
+
+  @action handleDrop(e) {
+    this.preventDefaults(e);
```
Comment (ankushdharkar) [inline app/components/upload-image.js:18]: Nit: Maybe add a comment here why?

File: app/components/upload-image.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      })
+      .catch((err) => {
+        this.setImageUploadSuccess(false);
+        this.setStatusMessage(
+          'Error occured, please try again and if the issue still exists contact administrator and create a issue on the repo with logs'
+        );
+        console.error(err);
+      })
+      .finally(() => {
+        this.setImageUploading(false);
```
Comment (ankushdharkar) [inline app/components/upload-image.js:86]: Awesome!

File: app/components/upload-image.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      });
+  }
+
+  updateFormData(file, key) {
+    const formData = new FormData();
+    formData.append(key, file);
+    this.formData = formData;
+  }
+
+  handelResponseStatusMessage(status, message) {
```
Comment (ankushdharkar) [inline app/components/upload-image.js:96]: *`handel`

File: app/router.js [hunk 8]
```diff
@@ -9,5 +9,6 @@ export default class Router extends EmberRouter {
 Router.map(function () {
   this.route('signup');
   this.route('notifications');
+  this.route('upload-image');
```
Comment (ankushdharkar) [inline app/router.js:12]: Can we make this `/profile` route please?

File: app/styles/image-upload.css [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  padding: 10px;
+  background: #ccc;
+  cursor: pointer;
+  font-size: 1.25rem;
+  border-radius: 5px;
+  border: 1px solid #ccc;
+  margin: 10px 0;
+}
+
+.image-form__button_disabled {
```
Comment (ankushdharkar) [inline app/styles/image-upload.css:43]: `.image-form__button--disabled`

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

**Review by ankushdharkar — APPROVED** [review]
Looks great! Thank you so much for doing this! 🎉

File: app/components/upload-image.hbs (hunk 2 above)
Comment (prakashchoudhary07) [reply app/components/upload-image.hbs:10]: https://www.w3.org/TR/html401/struct/text.html#h-9.3.1

**Pull request merged**